src/js/utils/BetaOptInUtil.js

```diff
@@ -0,0 +1,104 @@
+import Util from "./Util";
+
+const BETA_OPTIN_CONFIG_PATH = ["service"];
+const BETA_OPTIN_SCHEMA_PATH = ["properties", "service", "properties"];
+// SDK sets/reads this property within package config.
+const BETA_OPTIN_PROPERTY_NAME = "beta-optin";
+// Convert to dot notation for `Util.findNestedPropertyInObject`
+const DOT_NOTATION_SCHEMA_PATH = BETA_OPTIN_SCHEMA_PATH.concat(
+  BETA_OPTIN_PROPERTY_NAME
+).join(".");
+
+function getBetaProperty(configSchema) {
+  if (!Util.isObject(configSchema)) {
+    return false;
+  }
+
+  return Util.findNestedPropertyInObject(
+    configSchema,
+    DOT_NOTATION_SCHEMA_PATH
+  );
+}
+
+const BetaOptInUtil = {
+  /**
+   * Finds beta-optin property within cosmos package config schema.
+   *
+   * @param  {Object} configSchema - cosmos package config schema
+   * @return {Boolean} Returns true if configSchema contains beta-optin property
+   */
+  isBeta(configSchema) {
+    return getBetaProperty(configSchema) != null;
+  },
+
+  /**
+   * Get beta-optin property within config Object
+   *
+   * @param  {Object} configSchema - cosmos package config schema
+   * @return {Object} Returns property if found or null.
+   */
+  getProperty(configSchema) {
+    return getBetaProperty(configSchema);
+  },
+
+  /**
+   * Removes beta-optin property within config schema
+   *
+   * @param  {Object} configSchema - cosmos package config schema
+   * @return {Object} Returns new config Object without opt-in property
+   */
+  filterProperty(configSchema) {
+    const filteredConfig = Util.deepCopy(configSchema);
+
+    let keyIndex = 0;
+    let pointer = filteredConfig;
+
+    while (keyIndex < BETA_OPTIN_SCHEMA_PATH.length - 1) {
```

Comment:
Please always update a the variables of a loop condition right away to make sure that we don't create an infinite loop by accident.